plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraTable.java

```diff
@@ -0,0 +1,193 @@
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.lang.Math.min;
+import static java.lang.String.format;
+import static java.lang.String.join;
+import static java.util.stream.Collectors.joining;
+
+public class TestCassandraTable
+        extends TestTable
+{
+    private static final int TABLE_NAME_MAXIMUM_LENGTH = 48;
+
+    private final String tableName;
+
+    public TestCassandraTable(
+            SqlExecutor sqlExecutor,
+            CassandraServer server,
+            String keyspace,
+            String namePrefix,
+            List<ColumnDefinition> columnDefinitions)
+    {
+        this(sqlExecutor, server, keyspace, namePrefix, columnDefinitions, Optional.empty(), ImmutableList.of());
+    }
+
+    public TestCassandraTable(
+            SqlExecutor sqlExecutor,
+            CassandraServer server,
+            String keyspace,
+            String namePrefix,
+            List<ColumnDefinition> columnDefinitions,
+            Optional<String> tableOptions)
+    {
+        this(sqlExecutor, server, keyspace, namePrefix, columnDefinitions, tableOptions, ImmutableList.of());
+    }
+
+    public TestCassandraTable(
+            SqlExecutor sqlExecutor,
+            CassandraServer server,
+            String keyspace,
+            String namePrefix,
+            List<ColumnDefinition> columnDefinitions,
+            List<String> rowsToInsert)
+    {
+        this(sqlExecutor, server, keyspace, namePrefix, columnDefinitions, Optional.empty(), rowsToInsert);
+    }
+
+    public TestCassandraTable(
+            SqlExecutor sqlExecutor,
+            CassandraServer server,
+            String keyspace,
+            String namePrefix,
+            List<ColumnDefinition> columnDefinitions,
+            Optional<String> tableOptions,
+            List<String> rowsToInsert)
+    {
+        super(sqlExecutor, keyspace + "." + namePrefix, tableDefinition(columnDefinitions, tableOptions), columnNames(columnDefinitions), rowsToInsert);
+        this.tableName = getName().substring(keyspace.length() + 1);
+
+        try {
+            server.refreshSizeEstimates(keyspace, tableName);
+        }
+        catch (Exception e) {
+            try (TestCassandraTable ignored = this) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    public String getTableName()
+    {
+        return tableName;
+    }
+
+    public String getTableNameInKeyspace()
+    {
+        return getName();
+    }
+
+    @Override
+    protected String createTableName(String namePrefix)
+    {
+        // Override because Cassandra table names are limited to 48 characters in size
+        // https://cassandra.apache.org/doc/trunk/cassandra/cql/ddl.html#common-definitions
+        String tableNameInKeyspace = namePrefix + randomTableSuffix();
+        return tableNameInKeyspace.substring(0, min(tableNameInKeyspace.length(), tableNameInKeyspace.indexOf(".") + 1 + TABLE_NAME_MAXIMUM_LENGTH));
+    }
```

Comment:
We should instead opt to make table names shorter at the callsites where possible. Oracle also has this problem of being limited to 30 character names and we handle that by conciously choosing smaller table names.

IMO it's ok to rename some tables in the base test classes for this to work.

Reply:
> IMO it's ok to rename some tables in the base test classes for this to work.

Haha, I thought about doing this before, but I gave up because for newcomers may not know this rule, so the test will be wrong, which is not very friendly to newcomers.

But now that I think about it, this way is also problematic, not taking into account quoted names, such as `"test_space"." test_table"`, it seems better to let us follow a uniform rule compared to the cost of implementation. But we'd better add comment notes so that newcomers can mean this through the code.